pipeline/extractor/src/main/scala/org/sunbird/obsrv/extractor/functions/ExtractionFunction.scala

```diff
@@ -147,8 +156,10 @@ class ExtractionFunction(config: ExtractorConfig, @transient var dedupEngine: De
     wrapperEvent
   }
 
-  private def markBatchFailed(batchEvent: mutable.Map[String, AnyRef], error: Error): mutable.Map[String, AnyRef] = {
+  private def markBatchFailed(batchEvent: mutable.Map[String, AnyRef], error: Error, extractionKey: String): mutable.Map[String, AnyRef] = {
     super.markFailed(batchEvent, error, config.jobName)
+    if (extractionKey.nonEmpty && StringUtils.equals(extractionKey, Constants.EVENT))
+      batchEvent.remove(extractionKey)
```

Comment:
Why is extractionKey removed from batchEvent?

Reply:
We are remove the event from root level and adding as stringified event in the failed.

Following is the structure of failed event:
```
{
    "dataset-id": "dataset-id",
    "obsrv_meta": {
        "flags": { "ExtractorJob": "failed" },
        "syncts": 1696504444806,
        "prevProcessingTime": 1696504445231,
        "error": {
            "src": "ExtractorJob",
            "error_code": "ERR_EXT_1001",
            "error_msg": "Unable to extract the data from the extraction key"
        },
        "processingStartTime": 1696504445048,
        "timespans": { "ExtractorJob": 183 },
        "source": {
            "meta": {
                "id": "<dataset_source_config_id>", # For API this wouldn't be present
                "connector_type": "kafka/api/jdbc",
                "version": "1.0",
                "entry_source": <Table/Topic/API>
            },
            "trace_id": "<uuid>" # <uniquely_trace_event>
        }
    },
    "event": <STRINGIFIED JSON OF THE ENTIRE RECEIVED MESSAGE>
}
```